### DISCUSSION TITLE
如何判断符号链接是否存在

### DISCUSSION THREAD

**Discussion by xqyjlj** [xmake-io/xmake#4502] [Q&A]
xmake的 exits 是会一直追溯符号链接的
比如我现在的符号链接 `a -> /bin/b `,`exits(a)`就会判断 `/bin/b`是否存在。 `/bin.b` 文件不存在， a就指向了一个非法的的地址，`exits(a)`就会返回假。但是怎么检测a是否存在呢

![image](https://github.com/xmake-io/xmake/assets/51477609/af641727-4fd1-424f-97ed-3010730daf8a)

![image](https://github.com/xmake-io/xmake/assets/51477609/df29c552-6dcc-4e5c-85f5-f0fa5079c420)

**Comment by waruqi** [comment]
os.islink or os.isfile
